suite-common/graph/src/graphDataFetching.ts

```diff
@@ -230,3 +236,128 @@ export const getMultipleAccountBalanceHistoryWithFiat = async ({
 
     return mergeMultipleFiatBalanceHistories(accountsWithFiatBalanceHistory) as FiatGraphPoint[];
 };
+
+/**
+ * Calculates received and sent values of each balance movement point.
+ */
+export const formatBalanceMovementsAmounts = (balanceMovements: AccountMovementHistory[]) =>
+    A.map(balanceMovements, balanceMovement => {
+        const sentTotal = Number(balanceMovement.sent);
+        const receivedTotal = Number(balanceMovement.received);
+
+        const sentToSelf = Number.isNaN(Number(balanceMovement.sentToSelf))
+            ? 0
+            : Number(balanceMovement.sentToSelf);
+
+        return {
+            date: balanceMovement.time,
+            payload: {
+                sent: Math.abs(sentTotal - sentToSelf),
+                received: Math.abs(receivedTotal - sentToSelf),
+            },
+        } as BalanceMovement;
+    });
+
+/**
+ * Groups adjacent balance movements together.
+ *
+ * @param groupingThreshold maximum distance of tho balance movements to be grouped together.
+ */
+export const groupBalanceMovements = (
+    balanceMovements: readonly BalanceMovement[],
+    groupingThreshold: number,
+) => {
+    const groups: BalanceMovement[][] = [];
+    let currentGroup: BalanceMovement[] = [];
+    balanceMovements.forEach(balanceMovement => {
+        if (
+            A.isEmpty(currentGroup) ||
+            balanceMovement.date - A.last(currentGroup)!.date < groupingThreshold
```

Comment:
Please remove the `!`. Shouldn't be problem because of previous check but I guess it's just more error prone if some changes are added and typescript won't scream because you're using the non-nullable assertion.

Reply:
Fixed in 90955f4fc by not using `A.last()` then the `!` operator is not necessary.